### PR TITLE
CC-55,56,57,67,70: Add functional test cases.

### PR DESCRIPTION
**Pull request by gupta-ashutosh5**

### CODE REVIEW

File: lib/test_cases/function/pull_request_file.js [hunk 1]
```diff
@@ -409,6 +409,241 @@ var codeCensorPullRequestFile = function(){
...
+    function pullRequestFileCheckLongFunctions(prFile) {
+        let result = [];
+        if (prFile.type === 'php' || prFile.type === 'module' || prFile.type === 'theme') {
+            let targetPrLine = undefined;
+            let isFunction = 0;
+            let countFunctionLines = 0;
+            let fnStartLine, fnEndLine = 0;
+            prFile.lines.forEach(function (currentPrLine) {
+                if (currentPrLine instanceof PrLine) {
+                    const pattern = new RegExp("^function", "gm");
```
Comment (purushotamrai) [inline lib/test_cases/function/pull_request_file.js]: function can have spaces in the start

File: lib/test_cases/function/pull_request_file.js [hunk 2]
```diff
@@ -409,6 +409,241 @@ var codeCensorPullRequestFile = function(){
...
+                    let content = currentPrLine.content;
+                    if (pattern.test(content)) {
+                        isFunction = 1;
+                        fnStartLine = currentPrLine.number;
+                        targetPrLine = currentPrLine;
+                    }
+                    if (isFunction) {
+                        if (content.includes("{")) {
```
Comment (purushotamrai) [inline lib/test_cases/function/pull_request_file.js:590]: does this takes care of local if/switch blocks?

File: lib/test_cases/function/pull_request_file.js [hunk 3]
```diff
@@ -409,6 +409,241 @@ var codeCensorPullRequestFile = function(){
...
+
+    /**
+     * Check for internal URLs.
+     */
+    function pullRequestFileCheckInternalUrls(prFile) {
+        let result = [];
+        if (prFile.type === 'js') {
+            prFile.lines.forEach(function (currentPrLine) {
+                if (currentPrLine instanceof PrLine) {
+                    const pattern = new RegExp("[a-z\\-0-9]+\\/");
```
Comment (purushotamrai) [inline lib/test_cases/function/pull_request_file.js]: Simply relying on this won't be sufficient, we can check if the life also has http, href or some relevant checks.

Comment (purushotamrai) [reply lib/test_cases/function/pull_request_file.js]: link, a tag ...or something similar

File: lib/test_cases/function/pull_request_file.js [hunk 4]
```diff
@@ -409,6 +409,241 @@ var codeCensorPullRequestFile = function(){
...
+     * Check for name tags in subscriber service.
+     */
+    function pullRequestFileCheckSubscriberServiceTags(prFile) {
+        let result = [];
+        if (prFile.type === 'services.yml') {
+            let targetPrLine = undefined;
+            let is_pattern_present = 0;
+            prFile.lines.forEach(function (currentPrLine) {
+                if (currentPrLine instanceof PrLine) {
+                    const pattern = new RegExp("[a-z]+_subscriber\:");
```
Comment (purushotamrai) [inline lib/test_cases/function/pull_request_file.js]: Probably using check for "\EventSubscriber\" in currentPrLine.content will give better results.

Comment (purushotamrai) [reply lib/test_cases/function/pull_request_file.js]: Screenshot 2020-01-09 at 00 36 43

File: lib/test_cases/function/pull_request_file.js [hunk 5]
```diff
@@ -409,6 +409,241 @@ var codeCensorPullRequestFile = function(){
...
+                    let content = currentPrLine.content;
+                    if (pattern.test(content)) {
+                        is_pattern_present = 1;
+                        targetPrLine = currentPrLine
+                    }
+                    if (is_pattern_present) {
+                        if (content.includes("tags:")) {
```
Comment (purushotamrai) [inline lib/test_cases/function/pull_request_file.js]: content - is of currentPrLine while tags: can be present on next line, are you sure this working?

File: lib/test_cases/function/pull_request_file.js [hunk 6]
```diff
@@ -409,6 +409,241 @@ var codeCensorPullRequestFile = function(){
...
+                            if (content.includes(word)) {
+                                targetPrLine = currentPrLine;
+                                if (!content.includes("Drupal.checkPlain")) {
+                                    assigned_variable = content.split("=")[0].trim().split(" ").pop();
+                                }
+                            }
+                        });
+                    }
+                    if (typeof (assigned_variable) !== 'undefined') {
+                        if (content.includes(assigned_variable) && content.includes("Drupal.checkPlain")) {
```
Comment (purushotamrai) [inline lib/test_cases/function/pull_request_file.js]: content being used over here for validation looks like will be working on currentPrLine content and not the subsequent ones.. while we will generally have the checks in next line or subsequent lines. Kindly validate this scenario.

File: lib/test_cases/function/pull_request_file.js (hunk 3 above)
Comment (purushotamrai) [reply lib/test_cases/function/pull_request_file.js]: url

**Comment by gupta-ashutosh5** [comment]
Hi @purushotamrai 
done the changes, please review

**Review by purushotamrai — APPROVED** [review]
Good Work

**Pull request merged**